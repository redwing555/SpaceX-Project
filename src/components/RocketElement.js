import React from 'react';

const RocketElement = () => (<h1> This is my Rocket</h1>);

export default RocketElement;
