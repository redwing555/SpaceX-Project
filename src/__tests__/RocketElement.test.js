import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketElement from '../components/RocketElement';

describe('Rocket Element Tests', () => {
  test('Check if Component Renders in DOM', () => {
    render(
      <Provider store={store}>
        <RocketElement />
      </Provider>,
    );
    expect(screen.queryByText(/Missions/)).toBeNull();
    expect(screen.queryByText(/Description/)).toBeNull();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketElement />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
