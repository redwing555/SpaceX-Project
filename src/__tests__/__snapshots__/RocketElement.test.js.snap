// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Element Tests renders correctly 1`] = `
<div>
  <ul />
</div>
`;
